test/ql/com/ddg/ddg.cc

```diff
@@ -1,13 +1,16 @@
-#include "ql/ir/compat/compat.h"
-#include "ql/ir/old_to_new.h"
 #include "ql/com/ddg/build.h"
-#include "ql/com/ddg/ops.h"
```

Comment:
I think the intent of placing tests in src was to distinguish between
- unit tests, or so-called unit tests, accompanying the source
- integration tests, in tests/ folder
which is a pretty reasonable organization to me. Unfrotunately in OpenQL it's not well done, there are a bit of integration tests everywhere and almost no unit test, which is very annoying to know the source of an issue and the intent of the code.

Normally the unit tests link against only the piece they test.
OpenQL is just a massive CMake library so it doesn't bring any benefit.
Unfortunately my CMake change didn't go through, see https://github.com/QuTech-Delft/OpenQL/pull/460 where I tried to split CMake targets.

Reply:
If we had smaller CMake targets, having multiple test executables wouldn't be an issue, I think.

Reply:
I'll have to dedicate some time to go through the comments in that other PR! 😄 

We could perfectly try to not mix unit and integration tests in the same binary. That's perfectly sensible. Having, for example, an `OpenQL_unit_test` and an `OpenQL_integration_test`.

What I'm more in favour of, maybe because I think it's more an standard way, is to separate tests from source code. But, again, I may be very wrong here.

I'll leave this conversation unresolved so that we can talk about it next week.

Reply:
I think with CMake it's indeed standard to separate source and tests, fair point.
Let's not discuss the tests problem too much in this PR anyway - it's devoted to sth else.

I'm OK with them moving, actually your tests/CMakeLists.txt is a lot cleaner than the previous testing section in the CMakeLists at the root